.github/workflows/reusable_benchmarks.yml

```diff
@@ -13,68 +30,130 @@ env:
-          # Windows doesn't recognize 'CMAKE_BUILD_TYPE', it uses '--config' param in build command to determine the build type
-          - os: ubuntu-latest
-            extra_build_option: '-DCMAKE_BUILD_TYPE=Release'
+        os: ['ubuntu-latest']
     runs-on: ${{matrix.os}}
 
     steps:
-      - name: Checkout
-        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
-        with:
-          fetch-depth: 0
-
-      - name: Install apt packages
-        if: matrix.os == 'ubuntu-latest'
-        run: |
-          sudo apt-get update
-          sudo apt-get install -y cmake libhwloc-dev libnuma-dev libtbb-dev
-
-      - name: Initialize vcpkg
-        if: matrix.os == 'windows-latest'
-        uses: lukka/run-vcpkg@5e0cab206a5ea620130caf672fce3e4a6b5666a1 # v11.5
-        with:
-          vcpkgGitCommitId: 3dd44b931481d7a8e9ba412621fa810232b66289
-          vcpkgDirectory: ${{env.BUILD_DIR}}/vcpkg
-          vcpkgJsonGlob: '**/vcpkg.json'
-
-      - name: Install vcpkg packages
-        if: matrix.os == 'windows-latest'
-        run: vcpkg install
-        shell: pwsh # Specifies PowerShell as the shell for running the script.
-
-      - name: Configure build
-        run: >
-          cmake
-          -B ${{env.BUILD_DIR}}
-          ${{matrix.extra_build_option}}
-          -DCMAKE_INSTALL_PREFIX="${{env.INSTL_DIR}}"
-          -DCMAKE_PREFIX_PATH="${{env.VCPKG_PATH}}"
-          -DUMF_BUILD_SHARED_LIBRARY=ON
-          -DUMF_BUILD_BENCHMARKS=ON
-          -DUMF_BUILD_BENCHMARKS_MT=ON
-          -DUMF_BUILD_TESTS=OFF
-          -DUMF_FORMAT_CODE_STYLE=OFF
-          -DUMF_DEVELOPER_MODE=OFF
-          -DUMF_BUILD_LEVEL_ZERO_PROVIDER=ON
-          -DUMF_BUILD_CUDA_PROVIDER=ON
-          -DUMF_BUILD_LIBUMF_POOL_DISJOINT=ON
-          -DUMF_BUILD_LIBUMF_POOL_JEMALLOC=ON
-
-      - name: Build UMF on Linux
-        if: matrix.os == 'ubuntu-latest'
-        run: cmake --build ${{env.BUILD_DIR}} -j $(nproc)
-
-      - name: Build UMF on Windows
-        if: matrix.os == 'windows-latest'
-        run: cmake --build ${{env.BUILD_DIR}} --config Release -j $Env:NUMBER_OF_PROCESSORS
-
-      - name: Run benchmarks
-        working-directory: ${{env.BUILD_DIR}}
-        run: ctest -V --test-dir benchmark -C Release
+    - name: Add comment to PR
+      uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+      if: ${{ always() && inputs.pr_no != 0 }}
+      with:
+        script: |
+          const pr_no = '${{ inputs.pr_no }}';
+          const url = '${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}';
+          const params = '${{ inputs.bench_script_params }}';
+          const body = `Compute Benchmarks run (with params: ${params}):\n${url}`;
+
+          github.rest.issues.createComment({
+            issue_number: pr_no,
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            body: body
+          })
+
+    - name: Checkout
```

Comment:
juuust to be super safe, we could download UMF here to a sub-dir (like `ur-repo` for UR)